hack/generator/pkg/functions/resource_conversion_function.go

```diff
@@ -0,0 +1,308 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package functions
+
+import (
+	"fmt"
+	"go/token"
+
+	"github.com/dave/dst"
+
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astbuilder"
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astmodel"
+)
+
+// ResourceConversionFunction implements conversions to/from our hub type
+// Existing PropertyAssignment functions are used to implement stepwise conversion
+//
+// Direct conversion from the hub type:
+//
+// func (r <receiver>) ConvertFrom(hub conversion.Hub) error {
+// 	source, ok := hub.(*<hub.Type>)
+// 	if !ok {
+// 		return fmt.Errorf("expected <hub.Type> but received %T instead", hub)
+// 	}
+// 	return person.AssignPropertiesFromPerson(source)
```

Comment:
Is `person` here supposed to be `r` or something? It's not clear where it came from. Same in the example below.

Reply:
If you want to use a placeholder for the `receiver` type, then maybe also show that same placeholder in the assignment functions?

`r.AssignPropertiesFrom<Receiver>(source)`?